### DISCUSSION TITLE
How to find a specific item in a response.

### DISCUSSION THREAD

**Discussion by mattnield** [postmanlabs/postman-flows#307] [Q&A] [answered]
If we have a `List` in a response to a request, what is the most effective way to retrieve a single item from that `List`?

Take the example data here:

```json
{
  "objects": [
    {
      "codename": "mercury",
      "type": "planet",
      "satellites": 0
    },
    {
      "codename": "venus",
      "type": "planet",
      "satellites": 0
    },
    {
      "codename": "earth",
      "type": "planet",
      "satellites": 1
    },
    {
      "codename": "mars",
      "type": "planet",
      "satellites": 0
    },
    {
      "codename": "jupiter",
      "type": "planet",
      "satellites": 80
    },
    {
      "codename": "saturn",
      "type": "planet",
      "satellites": 80
    },
    {
      "codename": "uranus",
      "type": "planet",
      "satellites": 27
    },
    {
      "codename": "neptune",
      "type": "planet",
      "satellites": 14
    }
  ]
}
```

If we want to know the number of satellites for `jupiter`, what do we do? We _could_ have a loop with a condition, but I wondered if there is a more succinct method. I thought `find` might help, but I've never actually managed to get that to work and can find any documentation for it :(

Using JSONPath it would be something like `$.objects[?(@.codename=='jupiter')].satellites`

**Comment by samay-v** [comment, marked as answer]
@mattnield Thanks for the feedback. 
We're adding 2 new blocks to flow in the next release that'll help you solve this use case:
1. Find One
2. Find All

I'll add an example flow here once they are added 👍🏽

**Reply by mattnield** [reply]
fantastic! thanks @samay-v !

**Reply by jorge-hermoso** [reply]
Any update on the example mentioned above @samay-v ?

**Reply by samay-v** [reply]
@jorges119 here's one with the above stated data https://www.postman.com/postman/workspace/postman-answers/flow/62cfa7e9019dac04fb433a5c
